.github/workflows/ci.yml

```diff
@@ -1,29 +1,86 @@
 name: Ambianic Functions API CI Test 
 
-# Controls when the action will run. Triggers the workflow on push or pull request
-# events but only for the master branch
-on: [pull_request]
+on:
+  # Trigger the workflow on push or pull request against main branch
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   build:
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-    - name: Git checkout 
-      uses: actions/checkout@v2
-
     - name: Install Node JS ${{ matrix.node-version }}
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    
-    - name: Install Newman dependency
-      run: npm install --prefix tests/postman/
 
-    - name: Run the API and Postman's tests
+    - uses: actions/checkout@v1
+    - name: Install application dependencies
+      run: npm install
+
+    - name: Generate OpenAPI specs
       run: |
-        cd tests/postman/ && newman run *
+        npm run generate-swagger && ./scripts.sh cleanup
+
+#    - name: Install Newman dependency
+#      run: npm install --prefix tests/
+
+    - name: Local function testing
+      run:  npm run netlify-dev & sleep 10 && npm run test:postman
+      env:
+        STRIPE_TEST_KEY: ${{ secrets.STRIPE_TEST_KEY }}
+        ENDPOINT: http://localhost:5050
+
+    # - name: Run tests against deployed preview
+    #   run:   ./scripts.sh wait-for-url https://deploy-preview-2--ambianic-serverless.netlify.app && cd ./tests && node newman.js 
+    #   env:
+    #     STRIPE_TEST_KEY: ${{ secrets.STRIPE_TEST_KEY }}
+    #     ENDPOINT: https://deploy-preview-${{github.event.number}}--ambianic-serverless.netlify.app/.netlify/functions
+
+    - name: Wait for successful preview deployment
+      uses: jakepartusch/wait-for-netlify-action@v1
+      id: get-netlify-url
+      with:
+        site_name: "ambianic-serverless"
+        max_timeout: 60
+     
+    - name: Run tests against deployed preview
+      run:  npm run netlify-dev & sleep 10 && npm run test:postman
       env:
-        CI: true
+        STRIPE_TEST_KEY: ${{ secrets.STRIPE_TEST_KEY }}
+        ENDPOINT: ${{ steps.get-netlify-url.outputs.url }}
+
+    - name: Run OpenAPI Tests and generate Code Coverage
+      run: cd openapi-docs/ && npm install && npm run test
+
+    - name: Create a static API documentation page from openapi specs
+      run: npm run generate-docs
+
+      # run this step only if its a push to the main branch
+    - name: Publish static API documentation to Github Pages
```

Comment:
@vickywane have you tested the API docs deployment? Can I see how it looks on your fork?

Reply:
It works perfectly well. [This](https://vickywane.github.io/ambianic-subscriptions.github.io/) was from the last deployment.

Reply:
@vickywane the generated doc example looks good, but there are still questions.

1. Why are the names prefixed with `mock-` ? The deployment in your fork should look exactly the way they would show in a production deployment after merge.
2. Where are the JS SDK docs and examples for the generated JS client SDK for the API?